crates/ruff_linter/src/checkers/ast/analyze/statement.rs

```diff
@@ -1658,6 +1658,15 @@ pub(crate) fn statement(stmt: &Stmt, checker: &mut Checker) {
             if checker.enabled(Rule::NonPEP695TypeAlias) {
                 pyupgrade::rules::non_pep695_type_alias(checker, assign_stmt);
             }
+            if let Some(value) = value.as_deref() {
+                if checker.enabled(Rule::HardcodedPasswordString) {
+                    flake8_bandit::rules::assign_hardcoded_password_string(
+                        checker,
+                        value,
+                        std::slice::from_ref(target),
+                    );
+                }
+            }
```

Comment:
I think we should switch the condition so that we only check when the rule is enabled
```suggestion
			if checker.enabled(Rule::HardcodedPasswordString) {
            	if let Some(value) = value.as_deref() {
                    flake8_bandit::rules::assign_hardcoded_password_string(
                        checker,
                        value,
                        std::slice::from_ref(target),
                    );
                }
            }
```

Reply:
done,

I was using `Rule::LambdaAssignment` as an example:
https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/src/checkers/ast/analyze/statement.rs#L1633